### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by vttranlina**

### CODE REVIEW

**Comment by vttranlina** [comment]
`io.cucumber` look very outdated, (from 2018)
the compile failed when I tried to update 2.4.0 -> 7.13.0, I'm not in it, so I keep 2.4.0
Maybe we should have a plan to update it or maybe drop it? (it only use in test classes)

File: pom.xml [hunk 1]
```diff
@@ -2496,15 +2497,20 @@
                 <artifactId>jaxb-api</artifactId>
                 <version>${jaxb.version}</version>
             </dependency>
+            <dependency>
+                <groupId>net.javacrumbs.json-unit</groupId>
+                <artifactId>json-unit</artifactId>
+                <version>>${javacrumbs.json-unit.version}</version>
```
Comment (chibenwa) [inline pom.xml:2503]: Typo `>>` ?

**Comment by chibenwa** [comment]
org.apache.james.webadmin.routes.VacationRoutesTest.postVacationUpdatesAll

Minor date formating changes

```
expected: Optional[2021-09-20T10:00Z[UTC]]
 but was: Optional[2021-09-20T10:00Z]
```

Is there a way to configure jackson to keep the timezone?

org.apache.james.backends.opensearch.OpenSearchIndexerTest.updateMessages

```
org.awaitility.core.ConditionTimeoutException: 
Assertion condition defined as a lambda expression in org.apache.james.backends.opensearch.OpenSearchIndexerTest 
expected: 1L
 but was: 0L within 10 seconds.
	at org.apache.james.backends.opensearch.OpenSearchIndexerTest.awaitForOpenSearch(OpenSearchIndexerTest.java:252)
	at org.apache.james.backends.opensearch.OpenSearchIndexerTest.updateMessages(OpenSearchIndexerTest.java:109)
Caused by: org.opentest4j.AssertionFailedError: 

expected: 1L
 but was: 0L
	at org.apache.james.backends.opensearch.OpenSearchIndexerTest.lambda$awaitForOpenSearch$9(OpenSearchIndexerTest.java:257)
```

Related?

org.apache.james.jmap.draft.crypto.SecurityKeyLoaderTest.loadShouldReturnAsymmetricKeysWhenRawPublicKey

```
The provided [org.bouncycastle.asn1.x509.SubjectPublicKeyInfo] pem type is not (yet) supported
```

Some bouncy castle magic required here.

**Comment by vttranlina** [comment]
> org.apache.james.backends.opensearch.OpenSearchIndexerTest.updateMessages


IMO we should keep opensearch-java 2.1.0 for easier, I don't know why 2.6.0 throw

**Comment by chibenwa** [comment]
Can you share the error?

**Comment by vttranlina** [comment]
> Can you share the error?

sorry it did not throw, the query at `org.apache.james.backends.opensearch.OpenSearchIndexerTest#updateMessages` line 109 does not return totalHit = 1 (it returns 0)

**Comment by chibenwa** [comment]
Ok create a separate ticket to upgrade OpenSearch please...

**Comment by Arsnael** [comment]
OpenSearch you might want to upgrade the docker container as well, maybe the API slightly changed?

I agree to open a ticket just for this though. I just checked on Opensearch doc : https://opensearch.org/docs/latest/clients/java/

It seems the latest java client update has an integrated transport class. We need to measure the impact of refactoring that would allow us to drop the rest client library. If it's not too bad, then we need perf tests as well to see if there is a gain or not. If the result is good enough, we could switch and drop the opensearch-rest-client dependency.

WDYT?

**Comment by chibenwa** [comment]
Agree.

Ticket please.

**Comment by Arsnael** [comment]
A;right give me a minute to create it :)

**Comment by vttranlina** [comment]
> org.apache.james.webadmin.routes.VacationRoutesTest.postVacationUpdatesAll
> 
> Minor date formating changes
> 
> ```
> expected: Optional[2021-09-20T10:00Z[UTC]]
>  but was: Optional[2021-09-20T10:00Z]
> ```
> 
> Is there a way to configure jackson to keep the timezone?
> 


It looks like the new result after the update is better 
This was intended since 2.15 https://github.com/FasterXML/jackson-modules-java8/pull/267

I think we should modify test cases for that (don't change the configuration ObjectMapper)

**Comment by chibenwa** [comment]
Ok  yes there is still the `Z` which means zone offset. Of for the.

File: pom.xml [hunk 2]
```diff
@@ -2294,7 +2295,7 @@
             <dependency>
                 <groupId>commons-io</groupId>
                 <artifactId>commons-io</artifactId>
-                <version>2.11.0</version>
+                <version>2.13.0</version>
```
[the remark this answers is not in the document]
Comment (vttranlina) [reply pom.xml:2298]: > Does commons-io bring UnsynchronizedBuffered Input/Output Streams?

yes

File: server/apps/distributed-app/docker-compose.yml [hunk 3]
```diff
@@ -44,7 +44,7 @@ services:
       - "15672:15672"
 
   s3:
-    image: zenko/cloudserver:8.2.6
+    image: zenko/cloudserver:latest-7.10.19
```
Comment (Arsnael) [inline server/apps/distributed-app/docker-compose.yml:47]: Hm are you sure about this one? we go from 8.2.6 to 7.10.19... Can you double check? It could be fixes on the v7 and we would just go one version back actually

File: backends-common/opensearch/pom.xml [hunk 4]
```diff
@@ -88,7 +88,7 @@
         <dependency>
             <groupId>org.opensearch.client</groupId>
             <artifactId>opensearch-rest-client</artifactId>
-            <version>2.4.0</version>
+            <version>2.8.0</version>
```
Comment (Arsnael) [inline backends-common/opensearch/pom.xml:91]: You forgot to revert this one too :)

Comment (vttranlina) [reply backends-common/opensearch/pom.xml:91]: the test cases still pass with only upgrade opensearch-rest-client, so I still keep it
Do you think we should do both opensearch-rest-client, opensearch-java same time?

File: server/apps/distributed-app/docker-compose.yml (hunk 3 above)
Comment (vttranlina) [reply server/apps/distributed-app/docker-compose.yml:47]: you are right, 

them hasn't stored all images on dockerhub
It should be `registry.scality.com/cloudserver/cloudserver:8.7.25`

tldr: https://github.com/scality/cloudserver/issues/4986

File: backends-common/opensearch/pom.xml (hunk 4 above)
Comment (Arsnael) [reply backends-common/opensearch/pom.xml:91]: Ok for keeping it. 

No the goal of upgrading opensearch-java is that it might allow us to drop opensearch-rest-client dependency for good :)

File: server/apps/distributed-app/docker-compose.yml (hunk 3 above)
Comment (Arsnael) [reply server/apps/distributed-app/docker-compose.yml:47]: Means they migrated to their own repo. As long as it's public, there is no issue switching to it IMO

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java [hunk 5]
```diff
@@ -133,7 +133,7 @@ private static String buildSpecificImageDiscriminator() {
 
     @SuppressWarnings("resource")
     public DockerCassandra() {
-        this("cassandra_3_11_10-" + buildSpecificImageDiscriminator(), AdditionalDockerFileStep.IDENTITY);
+        this("cassandra_3_11_15-" + buildSpecificImageDiscriminator(), AdditionalDockerFileStep.IDENTITY);
```
Comment (chibenwa) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java:136]: Why not switch to Cassandra 4.1.x (in another PR maybe?)

**Comment by vttranlina** [comment]
Ci error
```
 Not same digest as expected: expected <16073f10204751cd71d3b4ea93be2649> was <d41d8cd98f00b204e9800998ecf8427e> -> [Help 1]
```
I don't know why it does not match, 
my md5 is `16073f10204751cd71d3b4ea93be2649` 
image

**Comment by chibenwa** [comment]
The digest is not stable, I encountered that as well.

-> Open a ticket upstream on glowroot github?
-> We might temporarilly drop glowroot, like we did on 3.7.x

**Comment by vttranlina** [comment]
> The digest is not stable, I encountered that as well.

it looks like from `download-maven-plugin` when upgrade 1.6.8 -> 1.7.0, I reverted it and now waiting to check new ci state

**Comment by vttranlina** [comment]
Squash fixup & rebase master & add new commit

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java (hunk 5 above)
Comment (vttranlina) [reply backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java:136]: https://github.com/linagora/james-project/issues/4822

**Review by Arsnael — APPROVED** [review]

**Comment by vttranlina** [comment]
Ci Issue?
 
```
com.github.dockerjava.api.exception.NotFoundException: Status 404: {"message":"No such image: rabbitmq:3.12.1-management"}

```
The image is already available:: https://hub.docker.com/layers/library/rabbitmq/3.12.1-management

**Comment by Arsnael** [comment]
Hm indeed... Maybe something wrong with the VM docker or connection issue when trying to fetch the image?Let me restart your build and we will see
